utils/litemap/examples/language_names_hash_map.rs

```diff
@@ -0,0 +1,45 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// LiteMap is intended as a small and low-memory drop-in replacement for
+// HashMap. This example demonstrates how it compares to HashMap.
+
+#![no_main] // https://github.com/unicode-org/icu4x/issues/395
+
+icu_benchmark_macros::static_setup!();
+
+use icu_locid::subtags::Language;
+use icu_locid_macros::language;
+use std::collections::HashMap;
+
+const DATA: [(Language, &'static str); 11] = [
+    (language!("ar"), "Arabic"),
+    (language!("bn"), "Bangla"),
+    (language!("ccp"), "Chakma"),
+    (language!("en"), "English"),
+    (language!("es"), "Spanish"),
+    (language!("fr"), "French"),
+    (language!("ja"), "Japanese"),
+    (language!("ru"), "Russian"),
+    (language!("sr"), "Serbian"),
+    (language!("th"), "Thai"),
+    (language!("tr"), "Turkish"),
+];
+
+#[no_mangle]
+fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    icu_benchmark_macros::main_setup!();
+
+    let mut map = HashMap::new();
+    // https://github.com/rust-lang/rust/issues/62633 was declined :(
+    for (lang, name) in DATA.iter() {
+        map.insert(lang, name).ok_or(()).unwrap_err();
+    }
+
+    debug_assert_eq!(11, map.len());
+    debug_assert_eq!(Some(&&"Thai"), map.get(&language!("th")));
+    debug_assert_eq!(None, map.get(&language!("de")));
+
+    0
+}
```

Comment:
I don't think we should be adding a hashmap example to `litemap` crate.

If you need binaries for perf/mem testing they should live in the perf/mem benchmark crate.

Reply:
Agree in principle, but ICU4X memory and code size benchmarks are based on examples.

Reply:
That does not address the core issue, that the purpose of this example is not to measure ICU4X memory or code size, but stdlib alternative. I understand the convenience of storing such file alongside ICU4X example, but it feels like a bad, excuse my french, "directory architecture" since in Rust, `examples` directory is what people use to learn how a crate may be used. This example is not helpful for that and is just noise from that perspective.

I understand why we already sacrifice the readability and main purpose of `examples` directory for our secondary goal (use examples for measuring), but starting to add examples of files that *don't* use the crate, and hence, can't be used as an example of how to use the crate even if squinting, seems like another step away from the original purpose of the dir, and I'd prefer not to make such tradeoff.

Non blocking, just expressing opinion.

Reply:
I'm happy to see us establish another standard place to put these binaries that are not really examples, but I don't want to do that in this PR.  I would like us to first make sure that all the example files are fully working in CI, and the consider it a refactoring effort to move them somewhere else.

Reply:
I understand that what will follow can be categorized as "slippery slope fallacy" argument, but if we accept "examples" like this, I believe that we should be also fine with storing files in `examples` that are binaries using `rust_icu` alternatives of `icu4x` components and use it for comparison between `rust_icu` and `icu4x`. And I have quite large `icu4x vs rust_icu 1-1 matches` binaries in https://github.com/zbraniecki/intl-measurements/

Should I move them to examples of our components? If not, why?

Reply:
In the mean time, I do have a short comment at the top of the file explaining what it is for.

Reply:
Gotcha.
Suggestion for the file comment: "This example does not use litemap. The reader may compare the file to the litemap example to showcase analogous operations using HashMap and LiteMap."

Reply:
Why not create a `wasm_benches` folder and use explicit `[[example]]` keys in Cargo.toml to teach it where to find them?

A comment is also fine.

Reply:
Updated the comment.

I like the idea of using explicit `[[example]]` keys, though!  I'll file a follow up to discuss this.